src/api.js

```diff
@@ -1,23 +1,30 @@
-const api = {
-    doImport (data) {
-        try { 
-            window.application.$store.dispatch('importModel', {
-                clientWidth: document.getElementById('svg-grid').clientWidth,
-                clientHeight: document.getElementById('svg-grid').clientHeight,
-                data: JSON.parse(data)
-            });
-        }catch(err){
-            return false;
-        }
-        return true;
-    },
-
-    doExport () {
-        return window.application.$store.getters['exportData'];
-    },
-
-    setConfig (config) {
-        console.log(config);
-        return true;
+window.api = {
+  config: {},
+  doImport(data) {
+    try {
+      window.application.$store.dispatch('importModel', {
+        clientWidth: document.getElementById('svg-grid').clientWidth,
+        clientHeight: document.getElementById('svg-grid').clientHeight,
+        data: JSON.parse(data),
+      });
+    } catch (err) {
+      return false;
     }
+    return true;
+  },
+  doExport() {
+    return window.application.$store.getters['exportData'];
+  },
+  setConfig({
+    units = 'm',
+    showMapDialogOnStart = true,
+    online = true,
+    onChange = function change() {},
+  }) {
```

Comment:
I tried calling `api.setConfig()`, and got an error "Uncaught TypeError: Cannot match against 'undefined' or 'null'." It worked when I called `api.setConfig({})`.

I don't know enough about how js does pattern matching to recommend a solution, sorry.